.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,43 @@
+// For format details, see https:https://containers.dev/implementors/spec/. For config options, see the README at:
+// https://github.com/devcontainers/images/tree/main/src/base-debian
+{
+    "name": "Kubernetes with e2e testing image build and push",
+    "build": {
+        "dockerfile": "../build/build-image/Dockerfile",
+        "args": {
+            "KUBE_CROSS_IMAGE": "registry.k8s.io/build-image/kube-cross",
+            "KUBE_CROSS_VERSION": "v1.29.0-go1.21.3-bullseye.0"
+        }
+    },
+    // Setup the go environment and mount into the dev container at the expected location
+    "hostRequirements": {
+        "cpus": 4
+    },
+    // Copy over welcome message and install pyyaml
+    "onCreateCommand": "bash .devcontainer/setup.sh",
+    // for Kubernetes testing, suppress extraneous forwarding messages
+    "otherPortsAttributes": {
+        "onAutoForward": "silent"
+    },
+    "remoteUser": "root",
+    // Install utils, go, docker-in-docker, and other tools
+    // For more info about dev container Features see https://containers.dev/features
+    "features": {
```

Comment:
Because the `kube-cross` image already has all the tools needed (except for pyyaml) you can remove these features.